src/States.cpp

```diff
@@ -105,29 +102,9 @@ namespace ale {
 
   ale::State States::getState(double time, ale::interpolation interp) const {
     int lowerBound = ale::interpolationIndex(m_ephemTimes, time);
-    int interpStart;
-    int interpStop;
-
-    if (m_ephemTimes.size() <= 3) {
-      interpStart = 0;
-      interpStop = m_ephemTimes.size() - 1;
-    }
-    else if (lowerBound == 0) {
-      interpStart = lowerBound;
-      interpStop = lowerBound + 3;
-    }
-    else if (lowerBound == m_ephemTimes.size() - 1) {
-      interpStart = lowerBound - 3;
-      interpStop = lowerBound;
-    }
-    else if (lowerBound == m_ephemTimes.size() - 2) {
-      interpStart = lowerBound - 2;
-      interpStop = lowerBound + 1;
-    }
-    else {
-      interpStart = lowerBound - 1;
-      interpStop = lowerBound + 2;
-    }
+    // try to copy the surrounding 8 points as that's the most possibly needed
+    int interpStart = std::max(0, lowerBound - 3);
+    int interpStop = std::min(lowerBound + 4, (int) m_ephemTimes.size() - 1);
```

Comment:
Only problem here is that at either the max or min index values you'll only get 4 values (specifically at when lowerBound is 0 or m_ephemTimes.size() - 1). I'm trying to think of doing something different but I can't think of anything right now.

Reply:
Lagrange interpolation generally gets less stable as you start adding points that are very far away. I'd be hesitant to use 6 points if the actual interpolation time is between the first two. I.E. interpolate at 0.5 and use points 0, 1, 2, 3, 4, 5